lib/prorate/throttle.rb

```diff
@@ -75,42 +87,45 @@ def <<(discriminator)
     #   with a arbitrary ratio - like 1 token per inserted row. Once the bucket fills up
     #   the Throttled exception is going to be raised. Defaults to 1.
     def throttle!(n_tokens: 1)
-      discriminator = Digest::SHA1.hexdigest(Marshal.dump(@discriminators))
-      identifier = [name, discriminator].join(':')
-
-      redis.with do |r|
-        logger.debug { "Applying throttle counter %s" % name }
-        remaining_block_time, bucket_level = run_lua_throttler(
-          redis: r,
-          identifier: identifier,
-          bucket_capacity: limit,
-          leak_rate: @leak_rate,
-          block_for: block_for,
-          n_tokens: n_tokens)
-
-        if remaining_block_time > 0
-          logger.warn { "Throttle %s exceeded limit of %d in %d seconds and is blocked for the next %d seconds" % [name, limit, period, remaining_block_time] }
-          raise ::Prorate::Throttled.new(name, remaining_block_time)
+      @logger.debug { "Applying throttle counter %s" % @name }
+      remaining_block_time, bucket_level = run_lua_throttler(
+        identifier: identifier,
+        bucket_capacity: @limit,
+        leak_rate: @leak_rate,
+        block_for: @block_for,
+        n_tokens: n_tokens)
+
+      if remaining_block_time > 0
+        @logger.warn do
+          "Throttle %s exceeded limit of %d in %d seconds and is blocked for the next %d seconds" % [@name, @limit, @period, remaining_block_time]
         end
-        return limit - bucket_level # How many calls remain
+        raise ::Prorate::Throttled.new(@name, remaining_block_time)
       end
+
+      @limit - bucket_level # Return how many calls remain
     end
 
     def status
-      discriminator = Digest::SHA1.hexdigest(Marshal.dump(@discriminators))
-      identifier = [name, discriminator].join(':')
-
-      redis.with do |r|
-        is_blocked = redis_key_exists?(r, "#{identifier}.block")
-        return Status.new(is_throttled: false, remaining_throttle_seconds: 0) unless is_blocked
-
-        remaining_seconds = r.get("#{identifier}.block").to_i - Time.now.to_i
-        Status.new(is_throttled: true, remaining_throttle_seconds: remaining_seconds)
+      redis_block_key = "#{identifier}.block"
+      @redis.with do |r|
+        is_blocked = redis_key_exists?(r, redis_block_key)
```

Comment:
nitpick; conventionally this should read `blocked?`

Reply:
🤔  Why is that? I believe the `?` is by convention only with boolean-returning methods, but not with local variables